### ISSUE TITLE
Why do we need `Conv2dStaticSamePadding`

### ISSUE THREAD

**Issue by yueruchen** [lukemelas/EfficientNet-PyTorch#122]
If I do not need to transform the tensorflow pre-trained weight into this model, instead I train it from scratch. Can I just delete the `Conv2dStaticSamePadding`?

**Comment by lukemelas** [comment]
Yup! If you don't intend to export the model, you can delete `Conv2dStaticSamePadding`. The static padding is used when you want to export the model (i.e. to production or to mobile, with a fixed image size).

**Comment by Zygonie** [comment]
And why is the static version exportable but not the dynamic one ?

**Comment by lukemelas** [comment]
That's just a PyTorch thing -- you have to have a static graph to export.

**Comment by Zygonie** [comment]
yes I understand, but for an export, you can just give a dummy input tensor
and save weights. As it is fully convolutional, no matter the input image
size, the result should be the same ?
With ONNX you can also set dynamic axes for example

**Comment by lukemelas** [comment]
I think you're right (at least now with all the advances in the recent versions of PyTorch). I put in the static padding because right after this repo was released, a bunch of people asked for exporting, and it was the easiest way I knew of doing exporting at the time.  

Do you think you might be able to write up a Colab notebook that does the ONNX export with non-static padding? If it looks good, I can remove the static padding and we can integrate your notebook into the README.

**Comment by Zygonie** [comment]
Sure, on it

Le mar. 23 juin 2020 17:46, Luke Melas-Kyriazi <notifications@github.com> a
écrit :

**Comment by Vozf** [comment]
Any progress on this one?
